build/download_k8s_proxy.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import packageInfo from "../package.json";
+import fs from "fs";
+import request from "request";
+import { ensureDir, pathExists } from "fs-extra";
+import path from "path";
+import { noop } from "lodash";
+import { isLinux, isMac } from "../src/common/vars";
+
+class K8sProxyDownloader {
+  public version: string;
+  protected url: string;
+  protected path: string;
+  protected dirname: string;
+
+  constructor(version: string, platform: string, arch: string, target: string) {
+    this.version = version;
+    this.url = `https://github.com/lensapp/lens-k8s-proxy/releases/download/v${this.version}/lens-k8s-proxy-${platform}-${arch}`;
+    this.dirname = path.dirname(target);
+    this.path = target;
+  }
+
+  public async checkBinary() {
+    const exists = await pathExists(this.path);
+
+    if (exists) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public async download() {
+    if (await this.checkBinary()) {
+      return console.log("Already exists");
+    }
+
+    await ensureDir(path.dirname(this.path), 0o755);
+
+    const file = fs.createWriteStream(this.path);
+
+    console.log(`Downloading lens-k8s-proxy ${this.version} from ${this.url} to ${this.path}`);
+    const requestOpts: request.UriOptions & request.CoreOptions = {
+      uri: this.url,
+      gzip: true,
+      followAllRedirects: true,
+    };
+    const stream = request(requestOpts);
+
+    stream.on("complete", () => {
+      console.log("lens-k8s-proxy binary download finished");
+      file.end(noop);
+    });
+
+    stream.on("error", (error) => {
+      console.log(error);
+      fs.unlink(this.path, noop);
+      throw error;
+    });
+
+    return new Promise<void>((resolve, reject) => {
+      file.on("close", () => {
+        console.log("lens-k8s-proxy binary download closed");
+        fs.chmod(this.path, 0o755, (err) => {
+          if (err) reject(err);
```

Comment:
Do we want to `resolve` in the else back, to ensure `fs.chmod` is done before resolving this Promise?

Reply:
Also wouldn't it be better to use `node-fetch` here instead of `request`?

Reply:
Not very good excuse but this file is almost direct c&p from kubectl download. Maybe I should rewrite these using node-fetch (I think it should be relatively easy to do).

Reply:
We could also just do that in a different PR too